Data Structures/Array Implementation/ArrayImplementation(Insert,Delete,Linear Search).cpp

```diff
@@ -0,0 +1,84 @@
+//A programme for basic Array Implmentation - Insertion, Deletion, Search (Linear).
+
+#include <stdio.h>
+
+  int linearsearch(int value, int n, int array[]) {
+    //Function which performs a linear search and returns the index.
+    int p = -1;
+    for (int i = 0; i < n; i++) {
+      if (value == array[i]) {
+        p = i; //Value found in the array
+        break;
+      }
+    }
+    return p;
+  }
+int main() {
+  int task = 0, position, n;
+  int value;
+  int sample_array[1000];
+  //Input of sample array
+  printf("Enter number of elements in array\n");
+  scanf("%d", & n);
+  printf("Enter elements\n");
+
+  for (int c = 0; c < n; c++) {
+    scanf("%d", & sample_array[c]);
+  }
+
+  while (task != 4) {
+    //User manual
+    printf("Press the index of task to be performed : \n");
+    printf("1.Insert\n");
+    printf("2.Delete\n");
+    printf("3.Search an element(Linear Search)\n");
+    printf("4.Exit\n");
+
+    scanf("%d", & task);
+
+    if (task == 1) {
+      printf("Enter the value to be inserted :\n");
+      scanf("%d", & value);
+      printf("Enter the position where you want to insert : \n");
+      scanf("%d", & position);
+      for (int p = n - 1; p >= position - 1; p--) {
```

Comment:
Why not create an **insert** function so that code is more readable and takes into account reusability?